android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt

```diff
@@ -98,6 +88,44 @@ class DVCClient private constructor(
         }
     }
 
+    private val onPauseApplication = fun () {
+        Timber.d("Closing streaming event source connection")
+        eventSource.close()
+    }
+
+    private val onResumeApplication = fun () {
+        if (eventSource.state != ReadyState.OPEN) {
+            eventSource.close()
+            Timber.d("Restarting streaming event source connection")
+            initEventSource()
+            refetchConfig(false, null)
+        }
+    }
+
+    private fun initEventSource () {
+        eventSource = EventSource.Builder(Handler(fun (messageEvent: MessageEvent?) {
+            if (messageEvent != null) {
+                val data = JSONObject(messageEvent.data)
+
+                if (data.has("data")) {
+                    val innerData = JSONObject(data.get("data") as String)
+                    val lastModified = if(innerData.has("lastModified")) {
+                        (innerData.get("lastModified") as Long)
+                    } else null
+
+                    val type = if(innerData.has("type")) {
+                        (innerData.get("type") as String).toLong()
+                    } else ""
+
+                    if (type == "refetchConfig" || type == "") { // Refetch the config if theres no type
+                        refetchConfig(true, lastModified)
+                    }
+                }
+            }
+        }), URI(config?.sse?.url)).build()
```

Comment:
this should only happen if the url is set

Reply:
oh right, fixed!